### PR TITLE
ROX-11911: top 10 CI failures

### PR DESCRIPTION
**Pull request by gavin-stackrox**
## Description

Send a weekly report of top 10 CI failures of merge to main branch tests separated by group.

## Checklist
- [x] Investigated and inspected CI test results

### Here I tell how I validated my change

Sent to #acs-ci-slack-integration-testing channel and looks good!

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: scripts/ci/metrics.sh [hunk 1]
```diff
@@ -133,3 +133,86 @@ _EO_UPDATE_
...
+    local subject="${2:-Top 10 QA E2E Test failures for the last 7 days}"
+    local is_test="${3:-true}"
+
+    local sql
+    # shellcheck disable=SC2016
+    sql='
+SELECT
+  MAX(count) AS Failures,
+  Classname AS Suite,
+  Name AS `Case`
```
Comment (janisz) [inline scripts/ci/metrics.sh:148]: The previous version used double select to handle multiple test from a single suite. Now it's not necessary if we want to have names

Comment (janisz) [reply scripts/ci/metrics.sh:148]:
```
  -- count of test failures in last week
  SELECT
    COUNT(*) AS count,
    Classname AS Suite,
    Name
  FROM
    `acs-san-stackroxci.ci_metrics.stackrox_tests__extended_view`
  WHERE
    CONTAINS_SUBSTR(ShortName, "qa")
    AND Status = "failed"
    AND NOT IsPullRequest
    AND CONTAINS_SUBSTR(JobName, "master")
    AND DATE(Timestamp) >= DATE_SUB(DATE_TRUNC(CURRENT_DATE(), WEEK(MONDAY)), INTERVAL 1 WEEK)
  GROUP BY
    Name,
    Classname
  ORDER BY
    COUNT(*) DESC
LIMIT
  10;
  ```

| count |                      Suite | Name                                                                                                                                                |
|------:|---------------------------:|-----------------------------------------------------------------------------------------------------------------------------------------------------|
|    15 | ExternalNetworkSourcesTest | Verify two flows co-exist if larger network entity added first                                                                                      |
|    15 | ExternalNetworkSourcesTest | Verify connection to a user created external sources                                                                                                |
|    15 | ExternalNetworkSourcesTest | Verify flow stays to the smallest subnet possible                                                                                                   |
|    15 | ExternalNetworkSourcesTest | Verify flow re-maps to larger subnet when smaller subnet deleted                                                                                    |
|    12 |          ImageScanningTest | Quay registry and scanner supports token and/or robot credentials - quay registry+scanner with token and robot creds                                |
|    12 |          ImageScanningTest | Quay registry and scanner supports token and/or robot credentials - quay registry+scanner with token                                                |
|    12 |               VulnMgmtTest | Verify severities and CVSS sha256:481960439934084fb041431f27cb98b89666e1a0daaeb2078bcbe1209790368c ncurses-base MODERATE_VULNERABILITY_SEVERITY 5.3 |
|    12 |               VulnMgmtTest | Verify severities and CVSS sha256:74ee7a5d7a7172090162b1b5f8022b3b403b9f4ac677d325209c56483452f417 ncurses LOW_VULNERABILITY_SEVERITY 9.8           |
|    12 |           IntegrationsTest | Verify Quay Registry+Scanner integration - default config                                                                                           |
|    10 |                    SACTest | Verify GetSummaryCounts using a token with partial access receives partial results                                                                  |

Comment (gavin-stackrox) [reply scripts/ci/metrics.sh]: :+1:

**Comment by gavin-stackrox** [comment]
@BradLugo @janisz WDYT, is this ready to go live? See slacks from https://redhat-internal.slack.com/archives/C068WMY4SEQ/p1702668719910479 is how it would look.

File: .github/workflows/ci-failures-report.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        SLACK_CI_INTEGRATION_TESTING_WEBHOOK: ${{ secrets.SLACK_CI_INTEGRATION_TESTING_WEBHOOK }}
+        SLACK_ENG_DISCUSS_WEBHOOK: ${{ secrets.SLACK_ENG_DISCUSS_WEBHOOK }}
+      shell: bash
+      run: |
+        source scripts/ci/lib.sh
+
+        setup_gcp
+
+        # Avoid BQ init message in first report
+        bq --quiet --format=pretty query --use_legacy_sql=false 'select 1'
```
Comment (janisz) [inline .github/workflows/ci-failures-report.yml:33]: will dry run do the job?

File: scripts/ci/metrics.sh [hunk 3]
```diff
@@ -133,3 +133,76 @@ _EO_UPDATE_
...
+
+    local data
+    echo "Running query with job match name $job_name_match"
+    data="$(bq --quiet --format=pretty query --use_legacy_sql=false "$sql" 2> /dev/null)" || {
+        echo >&2 -e "Cannot run query:\n${sql}\nresponse:\n${data}"
+        exit 1
+    }
+
+    if [[ -z "${data}" ]]; then
+        data="No failures!"
```
Comment (janisz) [inline scripts/ci/metrics.sh:174]: I hope one day we will run this path :)

File: scripts/ci/metrics.sh [hunk 4]
```diff
@@ -133,3 +133,76 @@ _EO_UPDATE_
...
+        },
+		{
+			"type": "section",
+			"text": {
+				"type": "mrkdwn",
+				"text": "```\n\($data)\n```"
+			}
+		}
+	]
+}'
```
Comment (janisz) [inline scripts/ci/metrics.sh:205]: nit: could you fix formatting?

**Review by janisz — APPROVED** [review]

File: .github/workflows/ci-failures-report.yml (hunk 2 above)
Comment (gavin-stackrox) [reply .github/workflows/ci-failures-report.yml:33]: I tried it. `--dry_run` does not create the `.bigqueryrc` file, so without this first run the first report will still have "Welcome to BigQuery" in the output. https://github.com/stackrox/stackrox/actions/runs/7226259185/job/19691439321?pr=9036#step:4:32

File: scripts/ci/metrics.sh (hunk 3 above)
Comment (gavin-stackrox) [reply scripts/ci/metrics.sh:174]: Maybe over the recharge break (if we disable nightlies (and gorman))

**Comment by BradLugo** [comment]
Also resolved:

- [ROX-21191](https://redhat.atlassian.net/browse/ROX-21191)
- [ROX-21192](https://redhat.atlassian.net/browse/ROX-21192)
- [ROX-21193](https://redhat.atlassian.net/browse/ROX-21193)
- [ROX-21208](https://redhat.atlassian.net/browse/ROX-21208)

[ROX-21191]: https://redhat.atlassian.net/browse/ROX-21191?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ROX-21192]: https://redhat.atlassian.net/browse/ROX-21192?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ROX-21193]: https://redhat.atlassian.net/browse/ROX-21193?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ROX-21208]: https://redhat.atlassian.net/browse/ROX-21208?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Pull request merged**